raimitigations/utils/data_utils.py

```diff
@@ -86,3 +86,74 @@ def err_float_01(param, param_name):
         raise ValueError(
             f"ERROR: invalid value for parameter '{param_name}'. " + "Expected a float value between [0.0, 1.0]."
         )
+
+
+# -----------------------------------
+def _transform_ordinal_encoder_with_new_values(
+    encoder: object, df: Union[pd.DataFrame, np.ndarray]
+) -> Tuple[Union[pd.DataFrame, np.ndarray], dict]:
+    """
+    Encodes categorical data using an existing EncoderOrdinal mapping
+    as well as a complimentary mapping for new values. It creates a new mapping for
+    values unseen at fit time on top of the existing mapping.
+
+    :param encoder: an EncoderOrdinal object to use for the base mapping;
+
+    :param df: pandas dataframe containing new unmapped values to encode;
+
+    :return: the encoded dataframe, the corresponding inverse-encoding map.
+
+    :rtype: pd.DataFrame or np.ndarray, dictionary.
+    """
+    df_cpy = df.copy()
+    df_cpy.columns = df.columns.astype(str)
+    encoder_mapping = encoder.get_mapping()
+    new_mapping = encoder_mapping.copy()
+    inverse_map_dicts = {}
+
+    for col in encoder.col_encode:
+        max_encoding = max(encoder_mapping[col]["labels"])
+        for val in (df_cpy.loc[:, col]).unique():
```

Comment:
Just out of curiosity: is there a reason why you're using ``df_cpy.loc[:, col].unique()`` instead of ``df_copy[col].unique()``?